pallets/ethereum-xcm/src/mock.rs

```diff
@@ -183,11 +184,72 @@ parameter_types! {
 	pub ReservedXcmpWeight: Weight = u64::max_value();
 }
 
+#[derive(
+	Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Encode, Decode, Debug, MaxEncodedLen, TypeInfo,
+)]
+pub enum ProxyType {
+	NotAllowed = 0,
+	Any = 1,
+}
+
+impl InstanceFilter<Call> for ProxyType {
+	fn filter(&self, _c: &Call) -> bool {
+		match self {
+			ProxyType::NotAllowed => false,
+			ProxyType::Any => true,
+		}
+	}
+	fn is_superset(&self, _o: &Self) -> bool {
+		false
+	}
+}
+
+impl Default for ProxyType {
+	fn default() -> Self {
+		Self::NotAllowed
+	}
+}
+
+parameter_types! {
+	pub const ProxyCost: u64 = 1;
+}
+
+impl pallet_proxy::Config for Test {
+	type Event = Event;
+	type Call = Call;
+	type Currency = Balances;
+	type ProxyType = ProxyType;
+	type ProxyDepositBase = ProxyCost;
+	type ProxyDepositFactor = ProxyCost;
+	type MaxProxies = ConstU32<32>;
+	type WeightInfo = pallet_proxy::weights::SubstrateWeight<Test>;
+	type MaxPending = ConstU32<32>;
+	type CallHasher = BlakeTwo256;
+	type AnnouncementDepositBase = ProxyCost;
+	type AnnouncementDepositFactor = ProxyCost;
+}
+
+pub struct EthereumXcmEnsureProxy;
+impl xcm_primitives::EnsureProxy<AccountId32> for EthereumXcmEnsureProxy {
+	fn ensure_ok(delegator: AccountId32, delegatee: AccountId32) -> Result<(), &'static str> {
+		let f = |x: &pallet_proxy::ProxyDefinition<AccountId32, ProxyType, u64>| -> bool {
+			x.delegate == delegatee && (x.proxy_type == ProxyType::Any)
```

Comment:
A very tricky parameter is the `delay` here - which dictates that the proxy should always announce its intent before `delay` blocks. If unannoced the `proxy` will [fail](https://docs.rs/pallet-proxy/latest/src/pallet_proxy/lib.rs.html#248) with unannounced. Do you know if we already handle this gracefully?

Reply:
I will check

Reply:
Good catch. There is also a public function we can use in pallet-proxy to find the proxy. Will add both